en/ui-system/components/engine/ui-batch.md

```diff
@@ -1,19 +1,91 @@
+To batch render 2D renderable components, the following criteria are required:
 
-## Batching Method Description
+| Prerequisite | Description |
+| :-- | :--- |
+| Nodes are hosted in the same layer | Renderable components in different layers are unable to be batched as layers predetermine the render process. |
+| Renderable components are drawn with the same material | Cocos Creator instantiates materials after `uniforms` are set which prevents batching. **It is prominent for renderable components to be batched to share the same material.** <br>If a `uniform` is created for a custom material and the material is to be batched after said `uniform` is no longer being used, reassign the material to the component via interface `CustomMaterial`. |
+| Material instances share the same settings for `BlendState` and `DepthStencilState` | `DepthStencilState` is automatized by the engine to perform depth testing and create stencil buffers. It is primarily used to implement visual effects such as masking. In general, it is not necessary for users to modify its parameters. |
+| Vertex data are transferred in the same **buffer** (new in v3.4.1.) | Vertex data are automatized by the engine under most scenarios and require no manual management. For more information, please refer to the **MeshBuffer Guidelines** section below. |
+| Textures share the same source and sampler type | The most common issue that prevents batching is a mismatch between textures. For example, Sprites and Labels are unable to be batched due to different sampler types. It is recommended to use a general workflow to achieve optimal batching results in Cocos Creator. Please refer to the **Batching Workflow Guidelines** section below for more details. |
 
-Combined with the above description of batching conditions, we can use some methods to achieve better batching methods, but it should be noted that the rendering data collection of 2D rendering components is a node-tree-based rendering method, and some components will interrupt the batching. The three middleware components, TiledMap, Spine and DragonBones, follow their own internal batching mechanism, so the above components can't be batched and will interrupt the batching of other components, so users need to manage the node tree layout in modules to achieve better batching.
+## Batching Workflow Guidelines
+
+> **Note**: Cocos Creator renders components based on the hierarchy of the **Node Tree** structure. Batching is prone to be interrupted when encountering a node that is prohibited from batching, thus preventing the rest of the nodes in the tree to be batched. Users are encouraged to manage their Node Tree structure to achieve an optimized outcome. 
```

Comment:
... Batching is prone to be interrupted when encountering a node that doesn't follow the same prerequisite settings with previous one, thus it starts to create a new batch using its own prerequisite settings. Users are encouraged to manage their Node Tree structure to achieve an optimized outcome.